source/non_matching/quadrature_generator.cc

```diff
@@ -1525,6 +1537,92 @@ namespace NonMatching
   {
     return surface_quadrature;
   }
+
+  template <int dim>
+  template <class VectorType>
+  DiscreteQuadratureGenerator<dim>::DiscreteQuadratureGenerator(
+    const hp::QCollection<1> &quadratures1D,
+    const DoFHandler<dim> &   dof_handler,
+    const VectorType &        level_set,
+    const AdditionalData &    additional_data)
+    : QuadratureGenerator<dim>(quadratures1D, additional_data)
+    , reference_space_level_set(
+        std::make_unique<RefSpaceFEFieldFunction<dim, VectorType>>(dof_handler,
+                                                                   level_set))
+  {}
+
+
+
+  template <int dim>
+  template <bool level_dof_access>
+  void
+  DiscreteQuadratureGenerator<dim>::generate(
+    const TriaIterator<DoFCellAccessor<dim, dim, level_dof_access>> &cell)
+  {
+    Assert(reference_space_level_set != nullptr, ExcNotInitialized());
+    reference_space_level_set->set_active_cell(cell);
+    const BoundingBox<dim> unit_box = create_unit_bounding_box<dim>();
+    QuadratureGenerator<dim>::generate(*reference_space_level_set, unit_box);
```

Comment:
Here you can just write
```c++
this->generate(*reference_space_level_set, unit_box);
```

Reply:
This only works if we decide to give `DiscreteQuadratureGenerator<dim>::generate` a different name than the called base class function.

Reply:
Hmm, I thought it would be fine since the two functions do not take the same arguments, but your are right.